### PR TITLE
Add DeliverMin setter

### PR DESCRIPTION
**Pull request by wltsmrz**
https://github.com/ripple/rippled/commit/4dc573f195c943e4d43e4eff4e09bde18cf6178f
https://github.com/ripple/rippled/commit/7c2478480d90c3d843ba4aa24b329ad430913c2c

### CODE REVIEW

File: src/core/transaction.js [hunk 1]
```diff
@@ -1186,9 +1162,6 @@ Transaction.prototype.pathAdd = function(path) {
   if (!Array.isArray(path)) {
     throw new Error('Path must be an array');
   }
-  // if (this.getType() !== 'Payment') {
```
Comment (sublimator) [inline src/core/transaction.js]: Why were these removed?

File: src/core/transaction.js [hunk 2]
```diff
@@ -1037,26 +1027,14 @@ Transaction.prototype.rippleLineSet = function(options_) {
...
 Transaction.prototype.setQualityIn = function(quality) {
-  // if (this.getType() !== 'TrustSet') {
-    // throw new Error('TransactionType must be TrustSet to use QualityIn');
-  // }
-
   return this._setUInt32('QualityIn', quality);
 };
 
 Transaction.prototype.setQualityOut = function(quality) {
-  // if (this.getType() !== 'TrustSet') {
```
Comment (sublimator) [inline src/core/transaction.js]: This is now handled by the "field unknown for type" like messages in SerializedObject.from_json ?

File: src/core/transaction.js (hunk 1 above)
Comment (wltsmrz) [inline src/core/transaction.js]: Eventually we'll subclass Transaction so they'd be unnecessary... I doubt that we'll ever switch these on prior to that, so just removing some clutter

Comment (sublimator) [inline src/core/transaction.js]: Fair enough

File: src/core/transaction.js [hunk 3]
```diff
@@ -1129,14 +1099,20 @@ Transaction.prototype.setDestination = function(destination) {
...
-    // throw new Error('TransactionType must be Payment to use SendMax');
-  // }
-
   return this._setAmount('SendMax', send_max);
 };
 
 /**
+ * Set DeliverMin for Payment
+ *
+ * @param {String|Object} deliver min  amount
```
Comment (sublimator) [inline src/core/transaction.js]: jsdoc name doesn't match the function parameter

File: src/core/transaction.js (hunk 2 above)
Comment (wltsmrz) [inline src/core/transaction.js]: Yeah, the serialization stuff should catch it, but I added this (although it's not enabled) along with other transaction validation things in the event that someone doesn't use local_signing

File: src/core/transaction.js (hunk 3 above)
Comment (wltsmrz) [inline src/core/transaction.js]: Me fix

Comment (sublimator) [inline src/core/transaction.js]: Me Jane

**Comment by sublimator** [comment]
L+G+T+M
